grails-app/taglib/au/org/ala/bie/BieTagLib.groovy

```diff
@@ -185,31 +208,64 @@ class BieTagLib {
 
     /**
      * Mark a phrase with language, optionally with a specific language marker
+     * <p>
+     * This can be done two ways, one as a semantic tag.
      *
      * @attr text The text to mark
      * @attr lang The language code (ISO) (defaults to the request locale or the default locale)
      * @attr mark Mark the language in text (defaults to true)
      * @attr href Link to the text
+     * @attr tag Use a tag marker, if false, use "in *language*" (defaults to true)
      */
     def markLanguage = { attrs ->
         Locale defaultLocale = RequestContextUtils.getLocale(request)
         String text = attrs.text ?: ""
-        Locale lang = Locale.forLanguageTag(attrs.lang ?: defaultLocale.language)
+        Locale lang = buildLocale(attrs.lang)
         String href = attrs.href
-        boolean mark = attrs.mark ?: true
+        boolean mark = attrs.containsKey('mark') ? attrs.mark : true
+        boolean tag = attrs.containsKey('tag') ? attrs.tag : true
 
         out << "<span lang=\"${lang}\">"
         if (href)
             out << "<a href=\"${href}\" target=\"_blank\" class=\"external\">${text}</a>"
         else
             out << text
         if (mark && defaultLocale.language != lang.language) {
-            String name = languageName(lang.language)
-            out << "&nbsp;<span class=\"annotation annotation-language\" title=\"${lang}\">${name}</span>"
+            def name = languages[lang.language] ?: [code: lang.toLanguageTag(), name: lang.displayName]
+            if (tag) {
+                out << "&nbsp;<span class=\"annotation annotation-language\" title=\"${name.code}\">${name.name}</span>"
+            } else {
+                String inLabel = message(code: 'label.in', default: 'in')
+                out << "<span class=\"in-marker\">&nbsp;${inLabel}&nbsp;</span><span class=\"language-name\" title=\"${lang.language}\">${name.name}</span>"
+            }
         }
         out << "</span>"
     }
 
+    /**
+     * Show language information, including a link to the language definition, if available
+     *
+     * @attr lang The language code
+     * @attr format Format into HTML (truye by default)
+     * @attr default The default language name, if not found
```

Comment:
FYI, If any of these `attr`s are "required", you can enforce this by adding the `REQUIRED`  term in the JavaDoc and you'll get compile time checking for this in GSP calls to this taglib.

Reply:
Changed and added taglib test cases.